src/permission/config-permission-handler.js

```diff
@@ -0,0 +1,243 @@
+	}
+};
+
+utils.inherits( ConfigPermissionHandler, events.EventEmitter );
+
+/**
+ * Temporary, will only be here until #146 (https://github.com/deepstreamIO/deepstream.io/issues/146)
+ * is resolved
+ *
+ * @param   {[type]}   connectionData [description]
+ * @param   {[type]}   authData       [description]
+ * @param   {Function} callback       [description]
+ *
+ * @returns {Boolean}
+ */
+/* istanbul ignore next */
+ConfigPermissionHandler.prototype.isValidUser = function( connectionData, authData, callback ) {
+	callback( null, authData.username || 'open' );
+};
+
+/**
+ * Load a configuration file. This will either load a configuration file for the first time at
+ * startup or reload the configuration at runtime
+ *
+ * @todo expose this method via the command line interface
+ *
+ * @param   {String} path the filepath of the permission.json file
+ *
+ * @public
+ * @cli loadConfig
+ * @returns {void}
+ */
+ConfigPermissionHandler.prototype.loadConfig = function( path ) {
+	fs.readFile( path, 'utf8', this._onConfigLoaded.bind( this, path ) );
+};
+
+/**
+ * Validates and compiles a loaded config. This can be called as the result
+ * of a config being passed to the permissionHandler upon initialisation,
+ * as a result of loadConfig or at runtime
+ *
+ * @todo expose this method via the command line interface
+ *
+ * @param   {Object} config deepstream permissionConfig
+ *
+ * @public
+ * @cli useConfig
+ * @returns {void}
+ */
+ConfigPermissionHandler.prototype.useConfig = function( config ) {
+	var validationResult = configValidator.validate( config );
+
+	if( validationResult !== true ) {
+		this.emit( 'error', 'invalid permission config - ' + validationResult );
+		return;
+	}
+
+	this._config = configCompiler.compile( config );
+	this._ruleCache.reset();
+	this._ready();
+};
+
+/**
+ * Implements the permissionHandler's canPerformAction interface
+ * method
+ *
+ * This is the main entry point for permissionOperations and will
+ * be called for every incoming message. This method executes four steps
+ *
+ * - Check if the incoming message conforms to basic specs
+ * - Check if the incoming message requires permissions
+ * - Load the applicable permissions
+ * - Apply them
+ *
+ * @param   {String}   username the name of the connected user, as specified in isValidUser
+ * @param   {Object}   message  a parsed deepstream message
+ * @param   {Function} callback the callback to provide the result
+ * @param   {[Object]}   authData additional optional authData as passed to isValidUser
+ *
+ * @public
+ * @interface
+ * @returns {void}
+ */
+ConfigPermissionHandler.prototype.canPerformAction = function( username, message, callback, authData ) {
+	if( typeof message.data[ 0 ] !== STRING ) {
+		callback( 'invalid message', false );
+		return;
+	}
+
+	var ruleSpecification = rulesMap.getRulesForMessage( message );
+	var name = message.data[ 0 ];
+	var ruleData;
+
+	if( ruleSpecification === null ) {
+		callback( null, true );
```

Comment:
I think having root permissions is an important part of using the config handler, otherwise you would use the 'open' permission handler instead. Also be weird for like:
```
{
event: {
   '*': {
        ...
   }
}
}
```

Since records would always be true without explicitly mentioning it..